lib/http1.c

```diff
@@ -120,6 +120,8 @@ static void init_request(struct st_h2o_http1_conn_t *conn)
             h2o_buffer_dispose(&conn->req_body);
         h2o_dispose_request(&conn->req);
     }
+    if (conn->_unconsumed_request_size)
+        h2o_buffer_consume(&conn->sock->input, conn->_unconsumed_request_size);
```

Comment:
Would it make sense to move these two lines into the end of `if` block above, right after `h2o_dispose_request`?

Based on my understanding that the intent is to retain the input buffer until `h2o_dispose_request` completes, I think doing so would make the code more understandable.

Reply:
Thank you, how does c16322e look?